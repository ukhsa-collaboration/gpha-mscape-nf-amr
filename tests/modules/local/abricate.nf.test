nextflow_process {

    name "Test Process RUN_ABRICATE_DB"
    script "modules/local/abricate.nf"
    process "RUN_ABRICATE_DB"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = Channel.of(
                    tuple(
                        "climb-id",
                        file("${projectDir}/tests/test_data/ERR3077524_GCF_013349085.1.test_data.5k_reads.fq", checkIfExists: true),
                        "card"
                    )
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
