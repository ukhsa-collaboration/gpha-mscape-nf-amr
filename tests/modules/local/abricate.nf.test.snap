{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "climb-id",
                        "card",
                        "abricate_card_out.tsv:md5,4473c7cbe8f16d7c88089bc0b5e15149"
                    ]
                ],
                "abricate_results": [
                    [
                        "climb-id",
                        "card",
                        "abricate_card_out.tsv:md5,4473c7cbe8f16d7c88089bc0b5e15149"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-15T08:38:47.459156134"
    }
}